src/test/java/com/dept/firstAssigmentDept/service/CatalogItemsServiceImpTest.java

```diff
@@ -0,0 +1,244 @@
+    @InjectMocks
+    private CatalogItemsServiceImp catalogItemsService;
+    @Mock
+    private CatalogItemToCatalogItemDTO catalogItemToCatalogItemDTO;
+    @Mock
+    private CatalogItemDTOToCatalogItem catalogItemDTOToCatalogItem;
+
+    @Test
+    void getAllCatalogItems_shouldReturnCatalogItems() {
+
+        CatalogItem catalogItem1 = createCatalogItem(1L,"TestItem1","The new super TestItem", new BigDecimal(99.99));
+        CatalogItem catalogItem2 = createCatalogItem(2L,"TestItem2","The super TestItem2", new BigDecimal(19.99));
+        CatalogItem catalogItem3 = createCatalogItem(3L,"TestItem3","The super TestItem3", new BigDecimal(1999.99));
+
+        List<CatalogItem> catalogItems = new ArrayList<>(Arrays.asList(catalogItem1, catalogItem2, catalogItem3));
+
+        when(catalogItemsRepository.findAll()).thenReturn(catalogItems);
+
+        List<CatalogItemDTO> actualCatalogItems = catalogItemsService.getAllCatalogItems();
+
+        assertThat(actualCatalogItems, is(notNullValue()));
+        assertThat(actualCatalogItems.size(), equalTo(3));
+
+    }
+
+    @Test
+    void getItemById_shouldReturnCorrectItem() {
+
+        CatalogItem testCatalogItem = createCatalogItem(1L,"TestItem","The new super TestItem", new BigDecimal(99.99));
+        CatalogItemDTO catalogItemDTO = new CatalogItemDTO(1L,"TestItem","The new super TestItem", new BigDecimal(99.99),testCatalogItem.getImages(),testCatalogItem.getCategories());
+
+        when(catalogItemToCatalogItemDTO.convert(testCatalogItem)).thenReturn(catalogItemDTO);
+        when(catalogItemsRepository.findById(testCatalogItem.getId())).thenReturn(Optional.of(testCatalogItem));
+
+        CatalogItemDTO actualCatalogItem = catalogItemsService.getItemById(testCatalogItem.getId());
+
+        assertThat(actualCatalogItem, is(notNullValue()));
+        assertThat(actualCatalogItem.getName(), is(equalTo("TestItem")));
+    }
+
+    @Test
+    void getItemById_shouldThrowNotFoundException() {
+
+        final Long number = 1L;
+        when(catalogItemsRepository.findById(number)).thenReturn(Optional.empty());
+
+        assertThrows(NotFoundException.class, () -> catalogItemsService.getItemById(number),
+                "There is no item with this Id: " + number + ".");
+        assertThatThrownBy(() -> catalogItemsService.getItemById(number))
+                .isInstanceOf(NotFoundException.class)
+                .hasMessage("There is no item with this Id: " + number + ".");
+    }
+
+    @Test
+    void addCatalogItem_shouldAddCatalogItem() {
+
+        CatalogItem testCatalogItem = createCatalogItem(1L,"TestItem","The new super TestItem", new BigDecimal(99.99));
+        CatalogItemDTO catalogItemDTO = new CatalogItemDTO(1L,"TestItem","The new super TestItem", new BigDecimal(99.99),testCatalogItem.getImages(),testCatalogItem.getCategories());
+
+        when(catalogItemToCatalogItemDTO.convert(testCatalogItem)).thenReturn(catalogItemDTO);
```

Comment:
this lines are redundun they are equal to 
`CatalogItemDTO response = catalogItemDTO`